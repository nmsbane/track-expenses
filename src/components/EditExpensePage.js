import React from "react";
import { connect } from "react-redux";
import ExpenseForm from "./ExpenseForm";
import { editExpense, removeExpense } from "../actions/expenses";

export class EditExpensePage extends React.Component {
  onSubmit = expense => {
    console.log("updated", expense);
    this.props.editExpense(this.props.expense.id, expense);
    this.props.history.push("/"); //redirect to dashboard page
  };

  onRemove = e => {
    this.props.removeExpense({ id: this.props.expense.id });
    this.props.history.push("/");
  };

  render() {
    return (
      <div>
        <ExpenseForm expense={this.props.expense} onSubmit={this.onSubmit} />
        <button onClick={this.onRemove}>Remove</button>
      </div>
    );
  }
}

const mapDispatchToProps = dispatch => {
  return {
    editExpense: (id, expense) => {
      return dispatch(editExpense(id, expense));
    },
    removeExpense: ({ id }) => {
      return dispatch(removeExpense({ id }));
    }
  };
};

const mapStateToProps = (state, props) => {
  // props points to the props passed into the component
  return {
    expense: state.expenses.find(
      expense => expense.id === props.match.params.id
    )
  };
};
export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage);
