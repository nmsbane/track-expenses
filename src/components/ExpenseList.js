import React from "react";
import { connect } from "react-redux";

const ExpenseList = props => (
  <div>
    <h1>Expense List {props.expenses.length}</h1>
    {props.filters.text}
  </div>
);

const mapStateToProps = state => {
  return {
    expenses: state.expenses,
    filters: state.filters
  };
};

const ConnectedExpenseList = connect(mapStateToProps)(ExpenseList);

export default ConnectedExpenseList;
