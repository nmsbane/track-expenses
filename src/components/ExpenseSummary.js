import React from "react";
import numeral from "numeral";
import selectExpenses from "../selectors/expenses";
import selectExpensesTotal from "../selectors/expense-total";
import { connect } from "react-redux";

export const ExpenseSummary = props => {
  const expenseWord = props.expensesCount > 1 ? "expenses" : "expense";
  const expenseTotal = numeral(props.expensesTotal / 100).format("$0,0.00");

  return (
    <div>
      <h1>
        Viewing {props.expensesCount} {expenseWord} totalling {expenseTotal}
      </h1>
    </div>
  );
};

const mapStateToProps = state => {
  const visibleExpenses = selectExpenses(state.expenses, state.filters);
  return {
    expensesCount: visibleExpenses.length,
    expensesTotal: selectExpensesTotal(visibleExpenses)
  };
};

const ConnectedExpenseSummary = connect(mapStateToProps)(ExpenseSummary);

export default ConnectedExpenseSummary;
