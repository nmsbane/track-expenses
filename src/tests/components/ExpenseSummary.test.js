import React from "react";
import { ExpenseSummary } from "../../components/ExpenseSummary";
import { shallow } from "enzyme";

test("should render correct html for 1 expense", () => {
  const wrapper = shallow(
    <ExpenseSummary expensesCount={1} expensesTotal={25} />
  );
  expect(wrapper).toMatchSnapshot();
});

test("should render correct html for more than 1 expense", () => {
  const wrapper = shallow(
    <ExpenseSummary expensesCount={2} expensesTotal={250} />
  );
  expect(wrapper).toMatchSnapshot();
});
