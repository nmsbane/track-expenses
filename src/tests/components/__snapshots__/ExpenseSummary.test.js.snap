// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correct html for 1 expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totalling 
    $0.25
  </h1>
</div>
`;

exports[`should render correct html for more than 1 expense 1`] = `
<div>
  <h1>
    Viewing 
    2
     
    expenses
     totalling 
    $2.50
  </h1>
</div>
`;
